src/error.rs

```diff
@@ -0,0 +1,52 @@
+use crate::middleware::{Statement, StatementTmpl};
+
+pub type Result<T, E = Error> = core::result::Result<T, E>;
+
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    // Wrappers on top of other errors
+    #[error("std::io::Error: {0}")]
+    IOError(#[from] std::io::Error),
+    #[error("anyhow::Error: {0}")]
+    Anyhow(#[from] anyhow::Error),
```

Comment:
It would be interesting if we can distinguish plonky2 errors (which are `anyhow::Error`) from other `anyhow::Errors`).  But perhaps that would require a lot of boilerplate in the code :/